### DISCUSSION TITLE
How to deploy to android?

### DISCUSSION THREAD

**Discussion by garymmi** [snakers4/silero-vad#38] [Q&A] [answered]
How to deploy to android ?
1. how to do feature extraction on android
2. how to do inference on android

any c++ code reference?
thanks

**Comment by snakers4** [comment, marked as answer]
> how to do feature extraction on android

there is no need of any sort of feature extraction

model just consumes 16 kHz [-1, 1] normalized float values - https://github.com/snakers4/silero-vad/blob/master/utils_vad.py#L22-L38 `

normalize: bool = True` is the default in torchaudio

**Comment by snakers4** [comment]
> how to do inference on android

I assume you can use 

- this https://pytorch.org/mobile/android/
- or this https://github.com/microsoft/onnxruntime/blob/master/BUILD.md#Android / https://github.com/microsoft/onnxruntime

**Comment by snakers4** [comment]
> any c++ code reference?

no
we do not employ C++ engineers at the moment
you are welcome to share your Android recipes with the community

**Comment by snakers4** [comment]
I will keep this issue opened - https://github.com/snakers4/silero-vad/issues/37 - in case anyone would want to share their builds
